### ISSUE TITLE
Should we hide the taxonomic terms not filled out on the sides of our topic pages and sub-pages?

### ISSUE THREAD

**Issue by acorbi** [OpenDevelopmentMekong/wp-odm_theme#751]
Is it better for the user to see the structure even when they can’t click on most of the terms? The only site where most taxonomic terms are live is ODC. 

![screenshot from 2017-02-08 12-18-39](https://cloud.githubusercontent.com/assets/384894/22735133/bda5024e-edf8-11e6-8aba-19f056ca24dc.png)

**Comment by acorbi** [comment]
@prustar @gimmemochi @tthy @Huyeng ping

**Comment by jpeizer1** [comment]
Actually the only instance where *many* taxonomic terms are live is ODC, easily less than half are filled out and the % could be closer to only 1/3 even on ODC. I would also suggest the question be rephrased as "Should we hide the taxonomic terms not filled out on the sides of our topic pages and sub-pages that have information on them?" I think the combination of no data on a number of country  sites and these empty taxonomic entries give the user and donor a sense of a half empty store. So I would suggest hiding them there. I think it is a good idea for somewhere on a country site (in its own separate page) for all taxonomic terms to be listed with a request to help fill out the missing items as Andy suggested some months ago.  I just don't think showing a bunch of empty sub-topics on our Topic information pages that do have information on them is the best place.

**Comment by acorbi** [comment]
@jpeizer1 a bit long, but i changed the title...

**Comment by prustar** [comment]
I'm all for hiding the topics/sub-topics on the side bar when there is no content for these topics available. However agree that there needs to be somewhere that we list the available (even potentially empty) topics are accessible to people from somewhere to be informative. 

Currently on the HOME page under topics this is not evident that we actually have any other topic other than on water. Then when you select TOPIC on the home page the list is presented in no particular order and a bit jumbled and random. There needs to be some form of narrative to the topics we present and currently there isn't so we need to use these structures to present one for the user to explore. Even if it's presented in the current taxonomy hierarchy till we can curate this better.  

![image](https://cloud.githubusercontent.com/assets/14996905/22863711/c024ac72-f145-11e6-8e2a-c6c5a56defca.png)

![image](https://cloud.githubusercontent.com/assets/14996905/22863706/aa25eeb8-f145-11e6-9385-bbad0d73bafb.png)

I suggested a while ago a linkages to pages that would direct users up and down (back and forward) at the bottom of topic pages so that they could read content in a logical order if they choose to or select topics they are interested in on the side bar. -- @acorbi can you please find the original issue where this was previously raised?

**Comment by Huyeng** [comment]
Regarding hiding the taxonomic terms from navigation, ODC team is considering if the empty terms should be hidden or not. Most of ODC users are returning visitor (60%) who may already familiar with ODC's taxonomy, so reducing the terms may surprise to them since the site was launched about a year ago. 
For other reason, ODC will work on any content on public health, eg. the HIV/Aids, if a number of empty terms is hidden, there will be a question that if the only the HIV/AIDs is the priority health concern in Cambodia?

We understood that it is good to show those empty topics on the other country sites or Mekong. Thus, i would like to suggest to provide the option on the setting to allow the country to be able to hide or show the unclickable terms on their site. 

![hide-show-taxonomy](https://cloud.githubusercontent.com/assets/6613530/22921377/f12f17ee-f2cb-11e6-9075-b7fb9481f7bb.jpg)

**Comment by jpeizer1** [comment]
@Huyeng Actaully I just checked gooogle analytics and I think you may have the statistics reversed. Your new visitors to the new site are actually 68.93%. It was slightly lower for the old site at 61%. So most site visitors would not be familiar with the taxonomy. Still I think the idea of providing an option to hide or not is a good idea.

New ODC Site Sep 2016 - Feb 2017	
Sessions: 195,638	New Users: 134,863
New Visitor	 	134,863(68.93%)
Returning Visitor	60,775(31.07%)

Old Site Jan 2105 - Aug 2016
Sessions: 884,006	New Users: 542,415
New Visitor	 	542,451(61.36%)
Returning Visitor	341,555(38.64%)

**Comment by acorbi** [comment]
@DBishton @gimmemochi any thoughts?

**Comment by DBishton** [comment]
Hi all,

While the ODI team internally was agreed on hiding taxonomic terms in the context of ODM and the other new country sites, ODC has a user base that has now adjusted to the new site and the use of the taxonomy widget to navigate between topic pages. I think this discussion has demonstrated that there are different views on how best to improve this method of navigation, and while it would be ideal if each country site could toggle this functionality on or off, I think it is vital that core navigation features like this remain consistent between sites.

My view is that changing the way this operates requires a revamped means of navigating/UI concept that considers:

* including a site map feature, and/or taxonomic diagram, accessible from header or footer menu
* Exposing the custom 'category page' functionality of the JEO theme to allow navigation via the taxonomy  across content types
* How SDGs should be integrated into navigation/OD taxonomy mapping
* Feasibility of allowing navigating up and down categories as @prustar suggests

**Comment by jpeizer1** [comment]
@DBishton I think we need to separate the question of showing the entire taxonomy "somewhere" on the site, which we all seem to concur with, from showing the entire, mostly empty, taxonomy running down the side of our topic pages that do contain information.  

What precisely is the benefit of a user seeing a list 500+ topics/subtopics running down the right side of the Topic pages when the vast majority contain no information on 4/5ths of our country sites? It only serves to starkly demonstrates the amount of information _we don't have_ on information pages designed to show what we do have. It gives the tantalizing impression this  information will be filled out in due course when we know it won't be any time soon for the vast majority of topics.

You indicate "I think it is vital that core navigation features like this remain consistent between sites."  For what logical reason? 1) The topics showing up on the navigation bar are already different between countries 2) When you switch countries, navigation now takes you to the home page so there is no broken links affect. So what is the problem of having a toggle on/off switch as Huy Eng suggests?
I wouldn't even have an issue giving the user a choice of showing all topics or only topics with information as an option on the page (with the default being only those with info).

Separate from the decision above I'd agree it would be useful to have a site map feature/ taxonomic diagram on the footer menu where you could expose the custom category page functionality. How we map the SDG's is another issue but if we were to make them topics/sub-topics assumedly you'd also show the ones that exist.

**Comment by gimmemochi** [comment]
> My view is that changing the way this operates requires a revamped means of navigating/UI concept that considers:

including a site map feature, and/or taxonomic diagram, accessible from header or footer menu
Exposing the custom 'category page' functionality of the JEO theme to allow navigation via the taxonomy across content types
How SDGs should be integrated into navigation/OD taxonomy mapping
Feasibility of allowing navigating up and down categories as @prustar suggests
<
@acorbi @DBishton @jpeizer1 @prustar My understanding is below.

- [ ] Including a site map feature -- we all agree on this. @acorbi Let's open a feature request.

- [ ] Exposing **Category page** only after the search design is finalized as this has raised issues that relate to the way we organize the content (see @prustar suggestion on Gitter, which I agreed with)

- [ ] How SDGs should be integrated / presented --> @DBishton SDGs indicators will be mapped to our taxonomy. There will be a dedicated SDGs page (a design of which will be proposing to the team) where editorial content and navigations to data on SDGs are presented. I imagine we introduce a way to toggle content view by ODI Taxonomy OR the SDGs. This doesn't affect our decision to hide/show vacant taxonomic topics on the **Topics pages** and can discuss this later.

- [ ] Feasibility of allowing navigating up and down categories as @prustar suggests --> I don't know this means. This probably needs some investigation. 

In the interim, should we hid vacant taxonomic pages? 
Yes, for ODM. No, for ODC. We can hid it now OR wait until we finalize and agree on the search presentation, and subsequently the category page. 

@acorbi Is there anything else you would like feedback on or should we come back to this after the search and category page discussions are finalized?

**Comment by prustar** [comment]
@acorbi - This issue raises a number of feature requests and I would like to break them out

1. We need to create a browse view for the TOPIC page like a traditional folder structure: example in my windows explorer: 
![image](https://user-images.githubusercontent.com/14996905/31878636-c4d6c764-b800-11e7-9414-1179ac8a0584.png)

Example from MyLAFF: (not so pretty but functional!)
![image](https://user-images.githubusercontent.com/14996905/31879019-fe6134b4-b801-11e7-9729-9474e4430589.png)

2. The home page should display curated topics by editors to highlight new articles and features.

3. The taxonomic tree should ONLY DISPLAY CONTENT THAT IS AVAILABLE. This feature needs to be optional as ODC will be affected if the development of the browse doesn't allow for navigation to sub-tier pages. Unless you can overcome this issue then I recommend we proceed with this. 

4. Navigation UP and DOWN the taxonomic tree needs to be enabled. This is a simple BACK and FORWARD option at the bottom of each topic page which is linked via the tree to navigate through the taxonomy so that readers are able to read things in order rather than randomly as they currently are.

**Comment by acorbi** [comment]
@prustar 

> We need to create a browse view for the TOPIC page like a traditional folder structure: example in my windows explorer:

This has been implemented since version 2.0, a long while ago. It is called _breadcrumbs_ and look like this:
![screenshot from 2017-10-23 12-06-10](https://user-images.githubusercontent.com/384894/31883573-947c6c74-b7ea-11e7-9432-067bf9fcf584.png)
on https://opendevelopmentcambodia.net/topics/concessions/

Also, the taxonomic widget on the topic page's sidebar has a folder/tree structure already which allows users to expand/collapse nodes just as windows file explorer does:

![screenshot from 2017-10-23 12-08-11](https://user-images.githubusercontent.com/384894/31883650-da4dcb62-b7ea-11e7-9198-5cb69cb536d7.png)

Isn't this what you mean? Can you put some more detailed descriptions on a draft if not? How would you implement this differently? 

> 2. The home page should display curated topics by editors to highlight new articles and features.

this is also already possible. See **Specific list of posts (optional)** on https://wiki.opendevelopmentmekong.net/partners:configuring_frontend_2_0#odm_custom_posts_widget_in_detail 

> 3. The taxonomic tree should ONLY DISPLAY CONTENT THAT IS AVAILABLE. This feature needs to be optional as ODC will be affected if the development of the browse doesn't allow for navigation to sub-tier pages. Unless you can overcome this issue then I recommend we proceed with this.

Noted. I have opened a new separated issue https://github.com/OpenDevelopmentMekong/wp-odm_theme/issues/1213 and added to my list

> 4. Navigation UP and DOWN the taxonomic tree needs to be enabled. This is a simple BACK and FORWARD option at the bottom of each topic page which is linked via the tree to navigate through the taxonomy so that readers are able to read things in order rather than randomly as they currently are.

Noted. I have opened a new separated issue https://github.com/OpenDevelopmentMekong/wp-odm_theme/issues/1214 and added to my list

**Comment by prustar** [comment]
@acorbi not talking about the breadcrumbs or side widget. Talking about the TOPIC page and it's function as a BROWSE and navigation for users to explore the site. See proposal here: https://docs.google.com/document/d/1YEzd6cT_aVQ_NW5C4gsEUK18NiIFOvq_vyMzxmF86dM/edit#heading=h.ag4ja1wp0ts0 

Just suggestion. Would benefit from a designer who could put together this idea to make it actually look good. We needs something in between the images and a folder symbol that would be attractive and enticing for people.
